File: js/LinkedList/test/LinkedListSpec.js
import LinkedList  from '../src/LinkedList';

describe('General', () => {

  let ll;
  beforeEach(() => {
    ll = new LinkedList();
  });

  it('LinkedList should be defined', () => {
    expect(LinkedList).toBeDefined();
  });

  it('the version should be 0.1', () => {
    expect(ll.version).toEqual(0.1);
  });
});

describe('LinkedList', () => {
  let ll;
  beforeEach(() => {
    ll = new LinkedList();
  });

  it('a list can be empty', () => {
    expect(ll.isEmpty()).toBe(true);
  });

  it('add an element in first position', () => {
    ll.addFirst('Lorem');
    expect(ll.length()).toBe(1);
    expect(ll.head.value).toBe('Lorem');
    expect(ll.tail.value).toBe('Lorem');
  });

  it('add a second element in first position ', () => {
    ll.addFirst('Lorem');
    ll.addFirst('Ipsum');
    expect(ll.length()).toBe(2);
    expect(ll.head.value).toBe('Ipsum');
    expect(ll.tail.value).toBe('Lorem');
  });

  it('get an element in last position', () => {
    ll.addLast('Lorem');
    expect(ll.head.value).toBe('Lorem');
  });

  it('get an element in last position', () => {
    ll.addLast('Lorem');
    ll.addLast('Ipsum');
    expect(ll.head.value).toBe('Lorem');
    expect(ll.tail.value).toBe('Ipsum');
  });

  it('prints Array', () => {
    expect(ll.toArray()).toEqual([]);
    ll.addLast('Lorem');
    ll.addLast('Ipsum');
    ll.addFirst('Dolor');
    var expected = [
      'Dolor',
      'Lorem',
      'Ipsum'];
    expect(ll.toArray()).toEqual(expected);
  });

});

File: ruby/LinkedList/spec/linked_list_spec.rb
require 'spec_helper'

describe Node do
  before :each do
    @node = Node.new('Woo')
  end

  describe '#new' do
    context 'returns a Node object' do
      it { expect(@node).to be_an_instance_of Node }
    end
  end

  describe '#value' do
    context 'value should be defined' do
      it { expect(@node.value).to eql 'Woo' }
    end
  end
end

describe LinkedList do
  before :each do
    new_node = Node.new('First')
    @ll = LinkedList.new(new_node)
  end

  describe '#new' do
    context 'returns a LinkedList object' do
      it { expect(@ll).to be_an_instance_of LinkedList }
      it { expect(@ll.head.value).to eql 'First' }
    end
  end
end
